### ISSUE TITLE
Unhandled throttling error (or rather insufficient wait/retries) causes problem initializing clusters with many nodes with awsbatch

### ISSUE THREAD

**Issue by sibowsb** [aws/aws-parallelcluster#1852]
**Environment:**
 - pcluster version: 2.7.0
 - Configuration file (~/.parallelcluster/config):
```
[aws]
aws_region_name = us-east-2

[global]
cluster_template = default
update_check = true
sanity_check = true

[aliases]
ssh = ssh {CFN_USER}@{MASTER_IP} {ARGS}

[cluster default]
key_name = xxxxx
base_os = alinux2
scheduler = awsbatch
desired_vcpus = 0
vpc_settings = default
max_vcpus = 4096
custom_ami = ami-xxxxxxxxxx
s3_read_write_resource = arn:aws:s3:::*
additional_iam_policies = arn:aws:iam::aws:policy/AmazonS3FullAccess
master_instance_type = t2.large

[vpc default]
vpc_id = vpc-xxxxxxxxxx
master_subnet_id = subnet-xxxxxxxxxx
compute_subnet_id = subnet-xxxxxxxxxx
```

**Bug description and how to reproduce:**
When submitting a job with many nodes using awsbatch, a subset of nodes fails to initialize due to too frequent DescribeStack API calls (`An error occurred (Throttling) when calling the DescribeStacks operation (reached max retries: 4): Rate exceeded`).

I submitted job via `awsbsub -n 100 -p 1 -m 4000 -cf xxxx.sh`. The log of the main node (by calling `awsbout <jobid>#0`) is stuck at `Detected 89/100 compute nodes. Waiting for all compute nodes to start.` for about 10 minutes. Upon inspecting the logs of individual compute nodes (by calling `awsbout <jobid>#x` where x is the id of the node), I found that 11 of the 100 nodes have the following output:
```
2020-06-29T19:03:00+00:00: Job id: 25f5ccae-82f9-46dd-ae35-ba98d2393855#26
2020-06-29T19:03:00+00:00: Initializing the environment...
2020-06-29T19:03:00+00:00: Starting ssh agents...
2020-06-29T19:03:00+00:00: Agent pid 7
2020-06-29T19:03:00+00:00: Identity added: /root/.ssh/id_rsa (/root/.ssh/id_rsa)
2020-06-29T19:03:08+00:00: An error occurred (Throttling) when calling the DescribeStacks operation (reached max retries: 4): Rate exceeded
```

**Comment by lukeseawalker** [comment]
Hi @sibowsb,
I'm looking into this.

**Comment by lukeseawalker** [comment]
I have verified and reproduced the problem and I'm marking this as a bug.

I don't have an easy fix, but I can provide a workaround to remove the `DescribeStacks` operation.

1. start with a new cluster OR if you have already one running make sure there are no jobs running (and no hosts in the `awsbhosts` output)

2. through AWS console or AWS cli locate the S3 bucket named `parallelcuster-<cluster_name>-<hash_string>`

3. from this bucket download the `docker/artifacts.zip` S3 object and unzip the archive

4.  edit the content of the `scripts/entrypoint.sh` file, removing the line
`
_master_ip="$(aws --region "${PCLUSTER_AWS_REGION}" cloudformation describe-stacks --stack-name "${PCLUSTER_STACK_NAME}" --query "Stacks[0].Outputs[?OutputKey=='MasterPrivateIP'].OutputValue" --output text)"
`
and replacing it with
`
_master_ip="${MASTER_IP}"
`

5. recreate the archive `artifacts.zip` with the new content, making sure the archive file listing is
```
> unzip -l artifacts.zip

Archive:  artifacts.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  07-02-2020 16:51   alinux/
     1724  05-19-2020 12:44   alinux/Dockerfile
        0  07-02-2020 16:51   alinux2/
     1724  05-19-2020 12:44   alinux2/Dockerfile
      617  05-19-2020 12:44   build-docker-images.sh
      585  05-19-2020 12:44   buildspec.yml
        0  07-02-2020 16:51   scripts/
     5466  05-19-2020 12:44   scripts/generate_hostfile.sh
     2419  05-19-2020 12:44   scripts/mount_nfs.sh
     2601  05-19-2020 12:44   scripts/mount_efs.sh
     1716  07-02-2020 16:53   scripts/entrypoint.sh
      684  05-19-2020 12:44   upload-docker-images.sh
---------                     -------
    17536                     12 files
```

6. upload the archive back into the S3 bucket i.e. `parallelcuster-<cluster_name>-<hash>/docker/artifacts.zip`

7. through the AWS console or AWS cli restart the Codebuild project named `parallelcluster-<cluster_name>-build-docker-images-project`

8. after build succeed we can start using the cluster

**Comment by yzh5239** [comment]
@lukeseawalker  Thanks for the info. Could you elaborate on step 7?

> through the AWS console or AWS cli restart the Codebuild project named parallelcluster-<cluster_name>-build-docker-images-project

I am not sure how to do that through aws cli. Thanks in advance.

**Comment by lukeseawalker** [comment]
Hi @yzh5239,
have you tried with
```
aws codebuild start-build --region <your_region> --project-name parallelcluster-<cluster_name>-build-docker-images-project
```
?

**Comment by yzh5239** [comment]
That works! Thank you!!

**Comment by tilne** [comment]
This should be fixed with v2.10.1, which was released today. Please let us know if you run into anymore issues.
